src/Microsoft.AspNetCore.Blazor.Razor.Extensions/GenericComponentPass.cs

```diff
@@ -45,20 +47,112 @@ public void VisitExtension(ComponentExtensionNode node)
             private void Process(ComponentExtensionNode node)
             {
                 // First collect all of the information we have about each type parameter
+                //
+                // Listing all type parameters that exist
                 var bindings = new Dictionary<string, GenericTypeNameRewriter.Binding>();
                 foreach (var attribute in node.Component.GetTypeParameters())
                 {
                     bindings.Add(attribute.Name, new GenericTypeNameRewriter.Binding() { Attribute = attribute, });
                 }
 
+                // Listing all type arguments that have been specified.
+                var hasTypeArgumentSpecified = false;
                 foreach (var typeArgumentNode in node.TypeArguments)
                 {
+                    hasTypeArgumentSpecified |= true;
```

Comment:
That's odd. Why not `hasTypeArgumentSpecified = true;`?

Reply:
eh, this just seemed naturl to me. If you think it's clearer the other way I don't mind.